### PR TITLE
fixes issues with menubar in fullscreen

### PR DESCRIPTION
**Pull request by sbatten**
#56314 #56293 #56291 #56127

### CODE REVIEW

File: src/vs/workbench/electron-browser/workbench.ts [hunk 1]
```diff
@@ -1160,7 +1177,19 @@ export class Workbench extends Disposable implements IPartService {
 	isVisible(part: Parts): boolean {
 		switch (part) {
 			case Parts.TITLEBAR_PART:
-				return this.getCustomTitleBarStyle() === 'custom' && (!browser.isFullscreen() || this.menubarVisibility === 'visible' || this.menubarVisibility === 'toggle');
+				if (this.getCustomTitleBarStyle() !== 'custom') {
```
Comment (bpasero) [inline src/vs/workbench/electron-browser/workbench.ts]: @sbatten we never want the custom title bar on mac, or do we? So I would also hardcode to `return false` if `isMacintosh` and then check for the setting.

**Review by bpasero — APPROVED** [review]

File: src/vs/workbench/electron-browser/workbench.ts (hunk 1 above)
Comment (bpasero) [reply src/vs/workbench/electron-browser/workbench.ts]: Sorry ignore that comment, we want the custom title bar on macOS but not the menu, but both are in one part now so this makes sense.

Comment (sbatten) [reply src/vs/workbench/electron-browser/workbench.ts]: Custom titlebar is the default on macOS. The menu is what we never want.

**Comment by sergei-dyshel** [comment]
Somehow this doesn't work for me on Linux with i3 window manager.
I'm using vscode 1.30, native titlebar style and menu bar visibility set to "toggle". And still vscode always displays titlebar even when menu is not toggled.
Any ideas how I debug this?

**Comment by sergei-dyshel** [comment]
@sbatten 	@bpasero

**Comment by sbatten** [comment]
@sergei-dyshel if you haven't already, please open a new issue with a screenshot and version info from vs code. I will follow up investigation there.

**Pull request merged**